export.js

```diff
@@ -0,0 +1,161 @@
+            //grabs the digit from Importance Name field for the id
+            if (prop == "ImportanceId") {
+                objTemplate[prop] = valueArray[i][j].charAt(0);
+            }
+            else if (prop == "ReleaseId"){
+                objTemplate[prop] = currentReleases[valueArray[i][j]];
+            }
+            else if ((prop == "AuthorId") || (prop == "OwnerId")){
+                objTemplate[prop] = currentUsers[valueArray[i][j]];
+                console.log(valueArray[i][j]);
+            }
+            else if (prop == "RequirementTypeId"){
+                objTemplate[prop] = requirementType[valueArray[i][j]];
+            }
+            else if (prop == "StatusId"){
+                objTemplate[prop] = reqStatus[valueArray[i][j]];
+            }
+            else if (prop == "ComponentId"){
+                objTemplate[prop] = currentComponents[valueArray[i][j]];
+            }
+            else {
+                objTemplate[prop] = valueArray[i][j];
+            }
+            j++
+        }
+        objArray.push(cleanObject(objTemplate));
+    }
+    return objArray;
+}
+
+function postNew(toSend, artifact, reqNum) {
+    let id = $('#projects').val();
+    if (toSend[reqNum] && toSend[reqNum].hasOwnProperty(toIdString(artifact))) {
+        $("<p>RequirementId " + toSend[reqNum].RequirementId + " was not updated<p>").appendTo('#error-box');
+        postNew(toSend, artifact, (reqNum + 1));
+    }
+    else if (reqNum < toSend.length) {
+        $.ajax({
+            async: true,
+            method: "POST",
+            crossDomain: true,
+            contentType: "application/json",
+            dataType: "json",
+            url: `${userInfo.spiraUrl}services/v5_0/RestService.svc/projects/${id}/${artifact}${userInfo.auth}`,
+            data: JSON.stringify(toSend[reqNum]),
+            success: function (data, textStatus, response) {
+                returnId(data.RequirementId, artifact, reqNum);
+                $("<p>" + toSend[reqNum].Name + " sent successfully<p>").appendTo('#error-box');
+            },
+            error: function () {
+                $("<p>" + toSend[reqNum].Name + " failed to send<p>").appendTo('#error-box');
+                enableButtons();
+            }
+        }).done(function (data, textStatus, response) {
+            if (toSend[reqNum + 1] && toSend[reqNum + 1].hasOwnProperty(toIdString(artifact))) {
+                $("<p>RequirementId " + toSend[reqNum + 1].RequirementId + " was not updated<p>").appendTo('#error-box');
+                postNew(toSend, artifact, (reqNum + 2));
+            }
+            else {
+                postNew(toSend, artifact, (reqNum + 1));
+            }
+        })
+    } else {
+        enableButtons();
+    }
+}
+
+function customFieldObjCreate(valueArray){
+    let newArray = valueArray.filter((val) => val != "");
+    for (let i in newArray){
+        let cusObj = {};
+        let valType = "";
+        switch (customFieldNames[i].Type){
```

Comment:
consider putting your if statements inside the switch.
eg `case "Integer": if (!Number.isInteger(newArray[i])){ newArray[i] = null }; break;`
I would also put integer as a specific case and have string as the default - so that the default is not to enforce any checking. You will need to merge a list type with the integer type for this